drivers/SmartThings/zwave-switch/src/zwave-dual-switch/fibaro-walli-double-switch/init.lua

```diff
@@ -11,40 +11,44 @@
 -- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
+local st_device = require "st.device"
 
 local FIBARO_WALLI_DOUBLE_SWITCH_FINGERPRINT = {mfr = 0x010F, prod = 0x1B01, model = 0x1000}
 
 local function can_handle_fibaro_walli_double_switch(opts, driver, device, ...)
   return device:id_match(FIBARO_WALLI_DOUBLE_SWITCH_FINGERPRINT.mfr, FIBARO_WALLI_DOUBLE_SWITCH_FINGERPRINT.prod, FIBARO_WALLI_DOUBLE_SWITCH_FINGERPRINT.model)
 end
 
-local function endpoint_to_component(device, endpoint)
-  if endpoint == 2 then
-    return "switch1"
+local function generate_child_name(parent_label)
+  if string.sub(parent_label, -1) == '1' then
+    return string.format("%s2", string.sub(parent_label, 0, -2))
   else
-    return "main"
+    return string.format("%s 2", parent_label)
   end
 end
 
-local function component_to_endpoint(device, component)
-  if component == "switch1" then
-    return {2}
-  else
-    return {1}
+local function device_added(driver, device)
+  if device.network_type ~= st_device.NETWORK_TYPE_CHILD then
+    local name = generate_child_name(device.label)
+    local metadata = {
+      type = "EDGE_CHILD",
+      label = name,
+      profile = "metering-switch",
+      parent_device_id = device.id,
+      parent_assigned_child_key = string.format("%02X", 2),
+      vendor_provided_label = name
+    }
+    driver:try_create_device(metadata)
   end
```

Comment:
can't this be inherited from the parent driver?

Reply:
@greens parent driver uses different profile ("switch-binary"), this one uses "metering-switch".